crates/adapters/hyperliquid/src/execution/mod.rs

```diff
@@ -13,36 +13,336 @@
 //  limitations under the License.
 // -------------------------------------------------------------------------------------------------
 
-//! Hyperliquid execution adapter for Rust Trading Toolkit.
-//!
-//! This module contains the implementation of an execution adapter that provides
-//! connectivity to the Hyperliquid exchange for order management and trade execution.
-//!
-//! ## Overview
-//!
-//! The Hyperliquid execution adapter integrates with NautilusTrader's execution
-//! framework to provide:
-//!
-//! - Real-time order placement, cancellation, and modification
-//! - Order status updates and trade reporting
-//! - Position and account status monitoring
-//! - Risk management and order validation
-//!
-//! ## Components
-//!
-//! ### HyperliquidExecutionClient
-//!
-//! The main execution client handles all order management operations:
-//!
-//! - Connects to Hyperliquid's WebSocket and REST APIs
-//! - Translates between NautilusTrader and Hyperliquid order formats
-//! - Manages order state and provides real-time updates
-//! - Handles authentication and signature generation
-//!
-//! TODO: Implement HyperliquidExecutionClient
-//! TODO: Add order management operations
-//! TODO: Add WebSocket event handling
-//! TODO: Add authentication and signature handling
+use std::sync::Mutex;
+
+use anyhow::{Context, Result, bail};
+use nautilus_common::{
+    messages::execution::{
+        BatchCancelOrders, CancelAllOrders, CancelOrder, ModifyOrder, QueryAccount, QueryOrder,
+        SubmitOrder, SubmitOrderList,
+    },
+    runtime::get_runtime,
+};
+use nautilus_core::UnixNanos;
+use nautilus_execution::client::{ExecutionClient, base::ExecutionClientCore};
+use nautilus_model::{
+    accounts::AccountAny,
+    enums::OmsType,
+    identifiers::{AccountId, ClientId, Venue},
+    orders::Order,
+    types::{AccountBalance, MarginBalance},
+};
+use serde_json;
+use tokio::task::JoinHandle;
+use tracing::{debug, info, warn};
+
+use crate::{
+    common::{consts::HYPERLIQUID_VENUE, credential::Secrets},
+    config::HyperliquidExecClientConfig,
+    http::client::HyperliquidHttpClient,
+    websocket::client::HyperliquidWebSocketClient,
+};
+
+#[derive(Debug)]
+pub struct HyperliquidExecutionClient {
+    core: ExecutionClientCore,
+    config: HyperliquidExecClientConfig,
+    http_client: HyperliquidHttpClient,
+    ws_client: Option<HyperliquidWebSocketClient>,
```

Comment:
I don't think the `ws_client` needs to be optional, we can init at the same time as the HTTP client.

Reply:
Thanks, I'll also standardize this across the adapters in a follow-up.